src/Services/EncounterService.php

```diff
@@ -257,111 +261,134 @@ public function getEncounterForPatient($pid, $eid)
             $results = sqlInsert(
                 $sql,
-                [
-                $data["date"],
-                $data["onset_date"],
-                $data["reason"],
-                $data["facility"],
-                $data["pc_catid"],
-                $data["facility_id"],
-                $data["billing_facility"],
-                $data["sensitivity"],
-                $data["referral_source"],
-                $pid,
+                $query['bind']
+            );
+
+            addForm(
                 $encounter,
-                $data["pos_code"],
-                $data["external_id"],
+                "New Patient Encounter",
+                $results,
+                "newpatient",
+                $data['pid'],
                 $data["provider_id"],
-                $data["parent_enc_id"]
-                ]
-            ),
-            "newpatient",
-            $pid,
-            $data["provider_id"],
-            $data["date"]
-        );
-
-        if ($results) {
-            return $encounter;
+                $data["date"]
+            );
+
+            if ($results) {
+                $processingResult->addData(array(
+                    'encounter' => $encounter,
+                    'uuid' => UuidRegistry::uuidToString($data['uuid']),
+                ));
+            } else {
+                $processingResult->addProcessingError("error processing SQL Insert");
+            }
         }
+
+        return $processingResult;
     }
 
-    public function updateEncounter($pid, $eid, $data)
+    /**
+     * Updates an existing Encounter record.
+     *
+     * @param $puuid The patient identifier of particular encounter.
+     * @param $euuid - The Encounter identifier used for update.
+     * @param $data - The updated Encounter data fields
+     * @return ProcessingResult which contains validation messages, internal error messages, and the data
+     * payload.
+     */
+    public function updateEncounter($puuid, $euuid, $data)
     {
-        $facilityService = new FacilityService();
+        $processingResult = new ProcessingResult();
+
+        $isValidPatient = $this->encounterValidator->validateId('uuid', self::PATIENT_TABLE, $puuid, true);
+        if (is_array($isValidPatient)) {
+            $processingResult->setValidationMessages($isValidPatient);
+            return $processingResult;
+        }
+        $isValidEncounter = $this->encounterValidator->validateId('uuid', self::ENCOUNTER_TABLE, $euuid, true);
+        if (is_array($isValidEncounter)) {
+            $processingResult->setValidationMessages($isValidEncounter);
+            return $processingResult;
+        }
 
+        $puuidBytes = UuidRegistry::uuidToBytes($puuid);
+        $euuidBytes = UuidRegistry::uuidToBytes($euuid);
+        $data['pid'] = $this->getIdByUuid($puuidBytes, self::PATIENT_TABLE, "pid");
+        $data['encounter'] = $this->getIdByUuid($euuidBytes, self::ENCOUNTER_TABLE, "encounter");
```

Comment:
shouldn't be inserting the above 2 lines in data since don't want those updated. can just store them here as normal variables that you then use below in the `$query['bind']`